google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbortedTransactionInjector.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+/**
+ * Interface for simulating an aborted read/write transaction in test code. You should only use this
+ * class in test cases to verify that your code handles aborted exceptions correctly.
+ *
+ * <p>DO NOT USE IN PRODUCTION CODE.
+ */
+public interface AbortedTransactionInjector {
```

Comment:
This probably should be package private.

Reply:
No, for two reasons:
1. It is referenced in integration tests, which are in a different package.
2. The intention is that other (framework) developers should be able to use this to test any custom aborted transaction handling.